### PR TITLE
fix: don't allow . or .. in feature url

### PR DESCRIPTION
**Pull request by Tymek**
We do some validation on flag names, but there's some cases that slip through. These are some cases that we should handle better.

With `..` as a name, you can't go into the flag in Unleash and you can't activate any environments because the it is interpreted as "go up a level".

### CODE REVIEW

File: src/lib/routes/util.ts
Comment (Tymek) [inline src/lib/routes/util.ts:1]: this change applies to features, tags, projects, context fields and addons

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/hooks/useFeatureCreatedFeedback.ts
Comment (thomasheartman) [inline frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/hooks/useFeatureCreatedFeedback.ts:1]: Why is this part of this PR? Seems unrelated. I don't have the context to understand what this is about, I think.

File: src/lib/routes/util.ts [hunk 1]
```diff
@@ -13,7 +13,11 @@ export const customJoi = joi.extend((j) => ({
     },
     validate(value, helpers) {
         // Base validation regardless of the rules applied
-        if (encodeURIComponent(value) !== value) {
+        if (
+            encodeURIComponent(value) !== value ||
+            value === '..' ||
+            value === '.'
```
Comment (thomasheartman) [inline src/lib/routes/util.ts:19]: Yeh, this is good. Are there other cases we might have missed? 

I checked the docs for this function, and there's also `- _ ! ~ * ' ( )`:

image

Do we want to allow them? I guess there's no harm in it because they don't break urls the same way. Sure, let's go.

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/hooks/useFeatureCreatedFeedback.ts
Comment (Tymek) [reply frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/hooks/useFeatureCreatedFeedback.ts:1]: looks like I didn't start from main

Comment (Tymek) [reply frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/hooks/useFeatureCreatedFeedback.ts:1]: I'll fix it

**Review by thomasheartman — APPROVED** [review]
